### ISSUE TITLE
Geocoding not working in 3D Tiles Terrain Classification demo

### ISSUE THREAD

**Issue by lilleyse** [AnalyticalGraphicsInc/cesium#6908]
The 3D Tiles Terrain Classification demo uses the geocoder to set the initial view, but no longer works after https://github.com/AnalyticalGraphicsInc/cesium/pull/6876.

Is there anything that looks wrong about the code below?

```
var geocoder = viewer.geocoder.viewModel;
geocoder.searchText = 'Vienna';
geocoder.flightDuration = 0.0;
geocoder.search();
```

@mramato would you have a chance to look into this?

**Comment by mramato** [comment]
Good catch.  This is an easy one to fix.  Basically you are calling the geocoder before terrain is ready and we weren't correctly waiting on the ready promise in the new code I added.  This would (almost) never happen in normal usage because terrain initializes faster than the user performing a geocode.

I'll open a PR to fix it.
